datafusion/src/datasource/local.rs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::datasource::protocol_registry::ProtocolHandler;
+use crate::error::DataFusionError;
+use crate::error::Result;
+use parquet::file::reader::ChunkReader;
+use std::any::Any;
+use std::fs;
+use std::fs::{metadata, File};
+use std::sync::Arc;
+
+pub struct LocalFSHandler;
+
+impl ProtocolHandler<File> for LocalFSHandler {
+    fn as_any(&self) -> &dyn Any {
+        return self;
+    }
+
+    fn list_all_files(&self, root_path: &str, ext: &str) -> Result<Vec<String>> {
+        list_all(root_path, ext)
+    }
+
+    fn get_reader(&self, file_path: &str) -> Result<Arc<dyn ChunkReader<T = File>>> {
```

Comment:
Just my quick thoughts on possible designs to avoid the chunkreader trait here. We could declare a `read_file` method here instead that would take start and length as argument, then return a Vector or Iterator of bytes.

Then in `ParquetRootDesc`, we use a local struct to implement `ChunkReader` leveraging the `ObjectStore::read_file` method, which gets passed to `SerializedFileReader`.